simphony/io/tests/test_h5_mesh.py

```diff
@@ -106,5 +106,53 @@ def container_factory(self, name):
         group = self.handle.create_group(self.handle.root, name)
         return H5Mesh(group, self.handle)
 
+
+class TestH5MeshStoredLayout(unittest.TestCase):
+
+    def setUp(self):
+        self.temp_dir = tempfile.mkdtemp()
+        self.filename = os.path.join(self.temp_dir, 'test_file.cuds')
+        self.addCleanup(self.cleanup)
+        self.handle = tables.open_file(self.filename, mode='w')
+
+    def cleanup(self):
+        if os.path.exists(self.filename):
+            self.handle.close()
+        shutil.rmtree(self.temp_dir)
+
+    def test_new_mesh_layout(self):
+        # given
+        group = self.handle.create_group(self.handle.root, 'test')
+
+        # when
+        H5Mesh(group, self.handle)
+
+        # when
+        self.assertEqual(len(group._f_list_nodes()), 6)
+        self.assertIsInstance(group.cells, tables.Table)
+        self.assertIsInstance(group.points, tables.Table)
+        self.assertIsInstance(group.edges, tables.Table)
+        self.assertIsInstance(group.faces, tables.Table)
+        self.assertIsInstance(group.data, tables.Table)
+        self.assertIsInstance(group.item_data, tables.Table)
+
+    def test_mesh_layout_with_new_proxy(self):
```

Comment:
It is me (going and little overboard) checking that we do not create multiple nodes when we create multiple proxies of a 
already existing group. It is not directed for #148